### ISSUE TITLE
1.1 fails on Samsung S7 Edge

### ISSUE THREAD

**Issue by ianmacd** [janpascal/SubwayEdit#2]
The app fails to work on the Samsung S7 Edge, and also on the Samsung Galaxy S2 Tab. Both are running Android 6.0.1.

The logcat shows permission errors, so this is probably a Marshmallow issue.

I tried to fix the problem myself in Android Studio, but alas, I don't know my way around Android well enough yet.

It would be great if this app could be updated.

**Comment by janpascal** [comment]
Yeah, I'd noticed also on  another device running Marshmallow. I'll have a look when I find some round tuits

**Comment by ianmacd** [comment]
No worries.

Due to being too much of a beginner programmer on Android, I couldn't fix the permissions issue, in spite of a strong UNIX background.

Instead, I looked at your code and hacked up a few lines of Ruby to take a hand-edited playerdata file and write a new SHA1 hash and file size to it. It works for my quick purpose, but it's not a key/value editor, of course.

One thing I'm curious about: How did you figure out the salt to use with the SHA1 hash? I copied the array directly from your Java code.

In fact, how did you figure out the file format? I came across this project when searching for details of the format, and your spec of the header and body is the only one I could find anywhere on the web. Did you reverse-engineer it from scratch?

But I'm particularly curious how you figured out that salt. :-)

**Comment by janpascal** [comment]
I actually didn't figure it out, I found it somewhere on the web in a partially broken C implementation that, just like your Ruby code, fixes the hash off-line.

I guess the original author decompiled the Java class files inside the apk and worked from there.

For my part, the key-value parts of the playerdata file aren't very hard to reverse-engineer.

**Comment by janpascal** [comment]
Hi, in case you're still interested, I think I fixed the permissions issue with release 1.2. Could you please have a look if the tool now works for your device?
